src/schemas/listing-core.json

```diff
@@ -0,0 +1,243 @@
+    "moneyAmount": {
+      "type": "object",
+      "title": "An amount of money in a specified currency",
+      "properties": {
+        "currency": {
+          "type": "string"
+        },
+        "amount": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "currency",
+        "amount"
+      ]
+    },
+    "location": {
+      "type": "object",
+      "properties": {
+        "street": {
+          "type": "string"
+        },
+        "extendedStreet": {
+          "type": "string"
+        },
+        "locality": {
+          "type": "string"
+        },
+        "region": {
+          "type": "string"
+        },
+        "box": {
+          "type": "string"
+        },
+        "postalCode": {
+          "type": "string"
+        },
+        "country": {
+          "type": "string"
+        },
+        "coord": {
+          "type": "object",
+          "properties": {
+            "latitude": {
+              "type": "integer",
+              "minimum": -90,
+              "maximum": 90
+            },
+            "longitude": {
+              "type": "integer",
+              "minimum": -180,
+              "maximum": 180
+            }
+          },
+          "required": [
+            "latitude",
+            "longitude"
+          ]
+        }
+      },
+      "required": [
+        "street",
+        "locality",
+        "region",
+        "country"
+      ]
+    },
+    "media": {
+      "type": "object",
+      "title": "Representation of a media asset such as an image or a video",
+      "properties": {
+        "url": {
+          "type": "string",
+          "format": "uri"
+        },
+        "contentType": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "dimension": {
+          "type": "object",
+          "properties": {
+            "height": {
+              "type": "integer"
+            },
+            "width": {
+              "type": "integer"
+            }
+          },
+          "required": [
+            "height",
+            "width"
+          ]
+        }
+      }
+    },
+    "timeSlot": {
+      "type": "object",
+      "required": [
+        "startDate",
+        "endDate",
+        "isAvailable"
+      ],
+      "properties": {
+        "startDate": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "endDate": {
+          "type": "string",
+          "format": "date-time"
+        },
+        "isAvailable": {
+          "type": "boolean"
+        },
+        "price": {
+          "$ref": "#/definitions/moneyAmount"
+        }
+      },
+      "required": [
+        "startDate",
+        "endDate",
+        "isAvailable",
+        "price"
+      ]
+    },
+    "unitListing": {
+      "type": "object",
+      "properties": {
+        "unitsAvailable": {
+          "type": "integer"
+        },
+        "price": {
+          "$ref": "#/definitions/moneyAmount"
```

Comment:
@micahalcorn , we are [still using](https://github.com/OriginProtocol/origin-dapp/blob/59609cce764ad48d2f5a9e8769717c902dec428b/src/components/listing-create.js#L5) `react-json-schema` for rendering our forms, and I recall that it didn't handle anything too fancy like these json pointers. 

I get worried introducing such fanciness at the 11th hour... 😬 Your thoughts?

Reply:
I actually have not worked with `react-jsonschema-form` at all. But it looks like [there is some support](https://github.com/mozilla-services/react-jsonschema-form#schema-definitions-and-references).

Reply:
react-json-schema does support references to local definitions. See https://github.com/mozilla-services/react-jsonschema-form#schema-definitions-and-references

But it does *not* support anyOf - see https://github.com/mozilla-services/react-jsonschema-form#json-schema-supporting-status

Now, per previous comment, I was thinking the core listing schema would be used by origin-js  to do validation but DApp can use a different schema for rendering the listing creation form.

Reply:
Gotcha. Did not realize this was for validation. And good to learn about `react-jsonschema-form` being able to handle local definitions.

Reply:
Yeah we use `ajv` for the origin-js schemas, and it does handle refs. I was able to get it to work before.